viewer/viewer-awt/src/main/java/org/icepdf/ri/common/widgets/AbstractColorButton.java

```diff
@@ -66,13 +66,12 @@ public AbstractColorButton(Controller controller,
         Insets insets = dropDownArrowButton.getMargin();
         dropDownArrowButton.setMargin(new Insets(insets.top, 0, insets.bottom, 0));
 
-        String imageSize = ViewerPropertiesManager.getInstance().getPreferences().get(
-                ViewerPropertiesManager.PROPERTY_ICON_DEFAULT_SIZE, Images.SIZE_LARGE);
-        if (imageSize.equals(Images.SIZE_LARGE)) {
-            setPreferredSize(new Dimension(48, 32));
-        } else if (imageSize.equals(Images.SIZE_SMALL)) {
-            setPreferredSize(new Dimension(36, 24));
-        }
+        Images.IconSize imageSize = Images.getDefaultIconSizeOr (ViewerPropertiesManager.getInstance(), Images.IconSize.LARGE);
+        int h = Images.getHeightValueForIconSize (imageSize);

```

Comment:
I'm starting to think  maybe the height and width should be stored.   Maybe this should be Images.getIconSizeDimension().

Reply:
Maybe. I don't see why this particular button class needs to be rectangular rather than square &mdash; adding a getter for the icon width or the icon dimension wouldn't change much in this case, since most icons in the default pack are square and there are only very few exceptions; yet, `AbstractColorButton` wants to always be rectangular.

Thus, just assuming that the icon will be square seems to be "good enough"™ to me. We're already asking (by means of JavaDoc) anyone subclassing `IconPack` to provide square icons only.

Also note that either method, `getHeightValueForIconSize` or `getDimensionForIconsSize`, cannot be accurate unless we retrieve these values on a per-icon basis (since not even our default icon pack has exclusively square icons) which could require us reading the same bitmap multiple times.

Reply:
Related: Should I make those icons which are rectangular (e.g. `signature_*_lg.png`) square?

Reply:
I'm pretty sure the code in question is related to making sure the popup shows up in the correct location.  There might be something there as well for accounting for the dropdown arrow offset.  I'll try a few experiments but I think as you eluded to earlier the preferred size should just be a reflection of the the child components.

Reply:
I looked into this a little more and I think this code can be removed and the following added at the end of void setupLayout()

```
Dimension buttonPreferredSize = colorButton.getPreferredSize();
Dimension dropDownArrowPreferredSize = dropDownArrowButton.getPreferredSize();
setPreferredSize(new Dimension(
        buttonPreferredSize.width + dropDownArrowPreferredSize.width,
        buttonPreferredSize.height));
validate();
```

Reply:
So in this case, we don't need the height/size of the icon at all, do I understand you correctly? If so, do you still prefer a method retrieving the icon `Dimension` rather than the height only?

Reply:
I don't think the size will be needed at all, but I'l give it another review to double check.  I think the actual  size could be pulled as needed from other constructs as needed.